### PR TITLE
Add window controls

### PR DESCRIPTION
**Pull request by Jupi007**
## Pull request checklist

- [x] I run `flutter test --update-goldens` 

https://user-images.githubusercontent.com/36476595/201355492-8a4f5043-1141-4d3c-8333-eeb543de385f.mp4

https://user-images.githubusercontent.com/36476595/201355511-f71c37c2-8ad3-4f2c-883e-2427821af72c.mp4

### CODE REVIEW

File: lib/src/controls/yaru_window_control.dart [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  late CurvedAnimation _position;
+  late AnimationController _positionController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    oldType = widget.type;
+
+    _positionController = AnimationController(
```
Comment (jpnurmi) [inline lib/src/controls/yaru_window_control.dart:55]: missing dispose()?

**Review by jpnurmi — COMMENTED** [review]
So cool :D

File: lib/src/controls/yaru_window_control.dart (hunk 1 above)
Comment (Jupi007) [reply lib/src/controls/yaru_window_control.dart:55]: Oh, yes you're right thanks :)

**Review by Feichtmeier — APPROVED** [review]

**Comment by Feichtmeier** [comment]
The future is now! Great work

**Comment by Feichtmeier** [comment]
And how did you get peek running?

**Comment by Jupi007** [comment]
> And how did you get peek running?

Hmm, what do you mean? 🤔

**Comment by Feichtmeier** [comment]
((( the screen record tool does not work for me anymore under wayland, and I really miss it )))

**Comment by Jupi007** [comment]
Oh, okay.
Well, I'm also use Wayland, but I didn't do anything to get peek working, sorry.

**Comment by Jupi007** [comment]
I added a disabled state:

image

**Pull request merged**